gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -291,6 +291,12 @@ message ThrowErrorRequest {
   string errorCode = 2;
   // an optional error message that provides additional context
   string errorMessage = 3;
+  // JSON document that will instantiate the variables at the local scope of the
+  // job's associated task, and will propagate to upper scopes; it must be a JSON object, as variables will be mapped in a
```

Comment:
>// job's associated task, and will propagate to upper scopes;

❌ The variables should only be merged on the error catch event that catches the thrown error. It should not be propagated, unless output mappings are used to achieve this.